### PR TITLE
feat(metric_alerts): Make sure user conditions are enclosed in parentheses when building query to send to snuba

### PR DESCRIPTION
**Pull request by wedamija**
When we make metric alert related queries to snuba we modify the user query to include
`event.type:error` for queries with the `QueryDatasets.EVENTS` type. To make sure we support boolean
search properly, both this and the user provided query should be enclosed in parentheses.

Previously for a query like `release:123 OR release:456` we'd parse and send a query like `event.type:error and
release:123 OR release:456`, which gives us unexpected results. After this change, we'll send
`(event.type:error) and (release:123 OR release:456)`.

I also refactored how we combine the queries. It seems easier to combine our string query language
together rather than inject extra results into our parsed query structure.

Fixes WOR-316

### CODE REVIEW

File: tests/sentry/snuba/test_tasks.py [hunk 1]
```diff
@@ -158,27 +158,27 @@ def test_no_subscription_id(self):
 
 class BuildSnubaFilterTest(TestCase):
     def test_simple_events(self):
-        snuba_filter = build_snuba_filter(QueryDatasets.EVENTS, "", "count_unique(user)", None,)
+        snuba_filter = build_snuba_filter(QueryDatasets.EVENTS, "", "count_unique(user)", None)
```
Comment (EvanPurkhiser) [inline tests/sentry/snuba/test_tasks.py:161]: Do we need to run black again?

Or are you running a diff version of black dan 😬 ?

**Review by markstory — COMMENTED** [review]
Looks good to me. I agree that mutating the query string will be simpler than munging the parsed results. Using the string has the benefit of not needing to change when snql starts to come.

File: tests/sentry/snuba/test_tasks.py (hunk 1 above)
Comment (wedamija) [reply tests/sentry/snuba/test_tasks.py]: This should be fixed now.

**Review by iProgramStuff — APPROVED** [review]
Looks good to me!

**Pull request merged**